driver-core/src/main/com/mongodb/operation/AsyncQueryBatchCursor.java

```diff
@@ -73,6 +73,8 @@
     private volatile BsonDocument postBatchResumeToken;
     private volatile BsonTimestamp operationTime;
     private volatile boolean firstBatchEmpty;
+    private final AtomicBoolean noOperationInProgress = new AtomicBoolean(true);
```

Comment:
No need to be AtomicBooleans if access is synchronized against this.

Reply:
Changed to booleans.